source/common/quic/client_connection_factory_impl.cc

```diff
@@ -19,7 +19,6 @@ getContext(Network::TransportSocketFactory& transport_socket_factory) {
   auto* quic_socket_factory =
       dynamic_cast<QuicClientTransportSocketFactory*>(&transport_socket_factory);
   ASSERT(quic_socket_factory != nullptr);
-  ASSERT(quic_socket_factory->sslCtx() != nullptr);
```

Comment:
If this ASSERT were to fail the following code would crash.  If we remove the ASSERT, we should also fix this code so it doesn't crash.

https://github.com/envoyproxy/envoy/blob/1ea2aa686e096a64be4683aa7d72350bcb60b9d2/source/common/quic/envoy_quic_proof_verifier.cc#L35

Reply:
This is already removed over here https://github.com/envoyproxy/envoy/pull/16462
just hadn't merged it in.
That PR came with tests and error handling for SDS lazy loading of secrets :-)

Reply:
nit: Could we add ASSERT(context_ != nullptr) to the EnvoyQuicProofVerifier constructor as a sanity check?

https://github.com/envoyproxy/envoy/blob/main/source/common/quic/envoy_quic_proof_verifier.h#L15

Reply:
added in https://github.com/envoyproxy/envoy/pull/16466 thanks